homeassistant/components/evohome/__init__.py

```diff
@@ -119,13 +120,32 @@
 
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Create a (EMEA/EU-based) Honeywell TCC system."""
+    """Set up the Evohome integration from the configuration.yaml file."""
 
-    async def load_auth_tokens(store: Store) -> tuple[dict, dict | None]:
+    # until config flow is fully implemented, deleting any existing entry most closely
+    # matches the current behaviour
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(hass.config_entries.async_remove(entry.entry_id))
```

Comment:
The problem here is:
 - the entry could be disabled
 - the user may have changed the configuration.yaml after disabling

Reply:
So, I could do this:
```python
    if (entries := hass.config_entries.async_entries(DOMAIN)) and (entry := entries[0]):
        config[DOMAIN]["disabled_by"] = entry.disabled_by
        config[DOMAIN]["title"] = entry.title
        hass.async_create_task(hass.config_entries.async_remove(entry.entry_id))
    else:
        config[DOMAIN]["disabled_by"] = None
        config[DOMAIN]["title"] = None
```

Reply:
I would say yes. So update and reenable the entry on import,